Tools/DumpRenderTree/mac/ResourceLoadDelegate.mm

```diff
@@ -263,7 +271,16 @@ -(void)webView: (WebView *)wv resource:identifier didFinishLoadingFromDataSource
 
 -(void)webView: (WebView *)wv resource:identifier didFailLoadingWithError:(NSError *)error fromDataSource:(WebDataSource *)dataSource
 {
-    if (!done && gTestRunner->dumpResourceLoadCallbacks()) {
+    if (done) {
+        NSURL *failingURL = error.userInfo[@"NSErrorFailingURLKey"];
+        NSString* failingURLString = failingURL.absoluteString;
+        // Allow for an exact match, or a match with the trailing '/' missing from the argument URL
+        if ([mainResourceURLString isEqualToString:failingURLString]
+            || ([failingURLString hasPrefix:mainResourceURLString] && failingURLString.length - mainResourceURLString.length == 1)) {
```

Comment:
I was hoping there might be some kind of canonicalization methods—I didn't much care for this check otherwise. (And I noticed that WKURL didn't have this same problem with comparison.)